plugins/chanbackup.c

```diff
@@ -362,6 +495,237 @@ static struct command_result *json_state_changed(struct command *cmd,
 	return notification_handled(cmd);
 }
 
+
+static struct command_result *json_connect(struct command *cmd,
+                                           const char *buf,
+                                           const jsmntok_t *params)
+{
+	const jsmntok_t *nodeid = json_get_member(buf,
+                                                  params,
+                                                  "id");
+
+        struct node_id *node_id = tal(cmd, struct node_id);
+        json_to_node_id(buf, nodeid, node_id);
+        struct stat st;
+        struct out_req *req;
+
+	int fd = open("emergency.recover", O_RDONLY);
+
+        if (fd < 0)
+		plugin_err(cmd->plugin, "Opening: %s", strerror(errno));
+
+	if (stat("emergency.recover", &st) != 0)
+		plugin_err(cmd->plugin, "SCB file is corrupted!: %s",
+                           strerror(errno));
+
+	u8 *scb = tal_arr(cmd, u8, st.st_size);
+
+	if (!read_all(fd, scb, tal_bytelen(scb))) {
+		plugin_log(cmd->plugin, LOG_DBG, "SCB file is corrupted!: %s",
+                           strerror(errno));
+		return NULL;
+	}
+
+        u8 *serialise_scb = towire_peer_storage(cmd, scb);
+
+        req = jsonrpc_request_start(cmd->plugin,
+                                    cmd,
+                                    "sendcustommsg",
+                                    after_send_scb,
+                                    &forward_error,
+                                    node_id);
+
+        json_add_node_id(req->js, "node_id", node_id);
+        json_add_hex(req->js, "msg", serialise_scb,
+                     tal_bytelen(serialise_scb));
+
+	if (fsync(fd) != 0) {
+	        plugin_err(cmd->plugin, "closing: %s", strerror(errno));
+	}
+
+	if (close(fd) != 0) {
+	        plugin_err(cmd->plugin, "closing: %s", strerror(errno));
+	}
+
+        return send_outreq(cmd->plugin, req);
+}
+
+static struct command_result *after_datastore(struct command *cmd,
+                                             const char *buf,
+                                             const jsmntok_t *params,
+                                             void *cb_arg UNUSED)
+{
+        return command_hook_success(cmd);
+}
+
+
+static struct command_result *json_your_peer_storage(struct command *cmd,
+					         const char *buf,
+					         const jsmntok_t *params)
+{
```

Comment:
can we add a comment about the two types of custom messages that are valid here?

Reply:
Yes, I'll add comments and write a doc as well for this